Managers/PresetMan.cpp

```diff
@@ -291,24 +294,77 @@ int PresetMan::GetModuleID(string moduleName)
     return -1;
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////
+// Method:          GetModuleNameFromPath
+//////////////////////////////////////////////////////////////////////////////////////////
+// Description:     Gets the Name of a loaded DataModule, from a full data file path.
+std::string PresetMan::GetModuleNameFromPath( std::string dataPath )
+{
+    if( dataPath.empty() ) {
+        return "";
+    }
+
+    int slashPos = dataPath.find_first_of( '/' );
+    if( slashPos == string::npos ) {
+        slashPos = dataPath.find_first_of( '\\' );
+    }
+    std::string moduleName = dataPath.substr( 0, slashPos );
+    // check if path starts with Data/ or Mods/ and remove that part to get to the actual module name
+    if( moduleName == "Data" || moduleName == "Mods" ) {
+        std::string shortenPath = dataPath.substr( slashPos + 1 );
+        slashPos = shortenPath.find_first_of( '/' );
+        if( slashPos == string::npos ) {
+            slashPos = shortenPath.find_first_of( '\\' );
+        }
+        moduleName = shortenPath.substr( 0, slashPos );
+    }
+
+    return moduleName;
+}
 
 //////////////////////////////////////////////////////////////////////////////////////////
 // Method:          GetModuleIDFromPath
 //////////////////////////////////////////////////////////////////////////////////////////
 // Description:     Gets the ID of a loaded DataModule, from a full data file path.
 
-int PresetMan::GetModuleIDFromPath(std::string dataPath)
+int PresetMan::GetModuleIDFromPath( std::string dataPath )
 {
-    if (dataPath.empty())
+    if( dataPath.empty() ) {
         return -1;
+    }
 
-    int slashPos = dataPath.find_first_of('/');
-    if (slashPos == string::npos)
-        slashPos = dataPath.find_first_of('\\');
+    const std::string moduleName = GetModuleNameFromPath( dataPath );
+    return GetModuleID( moduleName );
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+// Method:          IsModuleOfficial
+//////////////////////////////////////////////////////////////////////////////////////////
+// Description:     Returns wether or not the module is vanilla.
 
-    return GetModuleID(dataPath.substr(0, slashPos));
+bool PresetMan::IsModuleOfficial(std::string moduleName)
+{
+	return std::find(allOfficialModules.begin(), allOfficialModules.end(), moduleName) != allOfficialModules.end();
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////
+// Method:          FullModulePath
+//////////////////////////////////////////////////////////////////////////////////////////
+// Description:     Returns the Full path to the module including Data/ or Mods/.
+// Arguments:       The Path to be completed.
+//                  The ID of the module to check.
```

Comment:
> The ID of the module to check.

I think this comment shouldn't be here anymore since it's probably vestigial from when the function received more arguments?